lib/Dialect/TritonGPU/Transforms/WSCodePartition.cpp

```diff
@@ -1245,18 +1444,11 @@ void buildAsyncComm(
 
     SmallVector<tt::ExperimentalDescriptorLoadOp> tmaLoads;
     SmallVector<Value> buffers;
+    DenseMap<Operation *, Operation *> producerCopyMap;
     // Go through all channels in this channel group.
     for (auto &c : kv.second) {
-      assert(
-          (isa<triton::LoadOp, tt::ExperimentalDescriptorLoadOp>(c->srcOp)) &&
-          "producer must be a LoadOp or tma LoadOp");
-      bool insideLoop = c->srcOp->getParentOfType<scf::ForOp>() != nullptr;
-      if (isa<triton::LoadOp>(c->srcOp)) {
-        // After createAsyncCopy, c->srcOp/headProducer are no longer valid.
-        createAsyncCopy(bufferMap, c, c->srcOp, asyncTasksPC, bufferIdx,
-                        bufferIdx);
```

Comment:
Oh lowering for non-tma loads are moved out of this function.

Reply:
Yeah as a part of the barrier insertion, as they are coupled.